modules/photo/include/opencv2/photo/ccm.hpp

```diff
@@ -0,0 +1,295 @@
+    COLOR_SPACE_Lab_A_10,                   ///< non-RGB color space
+    COLOR_SPACE_Lab_D55_2,                  ///< non-RGB color space
+    COLOR_SPACE_Lab_D55_10,                 ///< non-RGB color space
+    COLOR_SPACE_Lab_D75_2,                  ///< non-RGB color space
+    COLOR_SPACE_Lab_D75_10,                 ///< non-RGB color space
+    COLOR_SPACE_Lab_E_2,                    ///< non-RGB color space
+    COLOR_SPACE_Lab_E_10,                   ///< non-RGB color space
+};
+
+/** @brief Linearization transformation type
+*/
+enum LinearType
+{
+
+    LINEARIZATION_IDENTITY,                  ///<no change is made
+    LINEARIZATION_GAMMA,                      ///<gamma correction; Need assign a value to gamma simultaneously
+    LINEARIZATION_COLORPOLYFIT,               ///<polynomial fitting channels respectively; Need assign a value to deg simultaneously
+    LINEARIZATION_COLORLOGPOLYFIT,            ///<logarithmic polynomial fitting channels respectively; Need assign a value to deg simultaneously
+    LINEARIZATION_GRAYPOLYFIT,                ///<grayscale polynomial fitting; Need assign a value to deg and dst_whites simultaneously
+    LINEARIZATION_GRAYLOGPOLYFIT              ///<grayscale Logarithmic polynomial fitting;  Need assign a value to deg and dst_whites simultaneously
+};
+
+/** @brief Enum of possible functions to calculate the distance between colors.
+
+See https://en.wikipedia.org/wiki/Color_difference for details
+*/
+enum DistanceType
+{
+    DISTANCE_CIE76,                      ///<The 1976 formula is the first formula that related a measured color difference to a known set of CIELAB coordinates.
+    DISTANCE_CIE94_GRAPHIC_ARTS,         ///<The 1976 definition was extended to address perceptual non-uniformities.
+    DISTANCE_CIE94_TEXTILES,
+    DISTANCE_CIE2000,
+    DISTANCE_CMC_1TO1,                   ///<In 1984, the Colour Measurement Committee of the Society of Dyers and Colourists defined a difference measure, also based on the L*C*h color model.
+    DISTANCE_CMC_2TO1,
+    DISTANCE_RGB,                        ///<Euclidean distance of rgb color space
+    DISTANCE_RGBL                        ///<Euclidean distance of rgbl color space
+};
+
+/** @brief Core class of ccm model
+
+Produce a ColorCorrectionModel instance for inference
+*/
+class CV_EXPORTS_W ColorCorrectionModel
+{
+public:
+    /** @brief Color Correction Model
+
+        Supported list of color cards:
+        - @ref COLORCHECKER_Macbeth, the Macbeth ColorChecker
+        - @ref COLORCHECKER_Vinyl, the DKK ColorChecker
+        - @ref COLORCHECKER_DigitalSG, the DigitalSG ColorChecker with 140 squares
+
+        @param src detected colors of ColorChecker patches;\n
+                    the color type is RGB not BGR, and the color values are in [0, 1];
+        @param constcolor the Built-in color card
+    */
+    CV_WRAP ColorCorrectionModel(const Mat& src, ColorCheckerType constcolor);
```

Comment:
please use `int` for `constcolor`. Unfortunately not all bindings generators support enums for now.

Reply:
Also please use InputArray here and bellow. It allows to add non-cpu branches in future without API change.